pkg/cloudprovider/gcp.go

```diff
@@ -1,37 +1,248 @@
+				keepAliases = append(keepAliases, aliasIPRange)
+				continue
+			}
+			if assignedIP, _, err := net.ParseCIDR(aliasIPRange.IpCidrRange); err == nil && !assignedIP.Equal(ip) {
+				keepAliases = append(keepAliases, aliasIPRange)
+			}
+		}
+		networkInterface.AliasIpRanges = keepAliases
+		operation, err := g.client.Instances.UpdateNetworkInterface(g.project, g.parseZone(instance.Zone), instance.Name, networkInterface.Name, networkInterface).Do()
+		if err != nil {
+			return nil, err
+		}
+		opName = operation.Name
+		break
+	}
+	return GCPWaitInput{
+		opName: opName,
+		zone:   g.parseZone(instance.Zone),
+	}, nil
 }
 
-func (a *GCP) WaitForResponse(interface{}) error {
-	return nil
+func (g *GCP) WaitForResponse(requestObj interface{}) error {
+	gcpWaitInput, ok := requestObj.(GCPWaitInput)
+	if !ok {
+		return fmt.Errorf("error decoding GCP requestObj, object not of type: GCPWaitInput %#v", requestObj)
+	}
+	_, err := g.client.ZoneOperations.Wait(g.project, gcpWaitInput.zone, gcpWaitInput.opName).Do()
+	return err
+}
+
+func (g *GCP) GetNodeEgressIPConfiguration(node *corev1.Node) ([]*NodeEgressIPConfiguration, error) {
+	configs := []*NodeEgressIPConfiguration{}
+	instance, err := g.getInstance(node)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving instance associated with node, err: %v", err)
+	}
+	for _, networkInterface := range g.getNetworkInterfaces(instance) {
+		config := &NodeEgressIPConfiguration{
+			Interface: networkInterface.Name,
+		}
+		v4Subnet, v6Subnet, err := g.getSubnet(networkInterface)
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving the network interface subnets, err: %v", err)
+		}
+		config.IFAddr = ifAddr{}
+		if v4Subnet != nil {
+			config.IFAddr.IPv4 = v4Subnet.String()
+		}
+		if v6Subnet != nil {
+			config.IFAddr.IPv6 = v6Subnet.String()
+		}
+		config.Capacity = capacity{
+			IP: g.getCapacity(networkInterface),
+		}
+		configs = append(configs, config)
+		break
+	}
+	return configs, nil
+}
+
+func (g *GCP) getSubnet(networkInterface *google.NetworkInterface) (*net.IPNet, *net.IPNet, error) {
+	var v4Subnet, v6Subnet *net.IPNet
+	region, subnet := g.parseSubnet(networkInterface.Subnetwork)
+	subnetResult, err := g.client.Subnetworks.Get(g.project, region, subnet).Do()
+	if err != nil {
+		return nil, nil, err
+	}
+	if subnetResult.IpCidrRange != "" {
+		_, v4Subnet, _ = net.ParseCIDR(subnetResult.IpCidrRange)
+	}
+	if subnetResult.Ipv6CidrRange != "" {
+		_, v6Subnet, _ = net.ParseCIDR(subnetResult.Ipv6CidrRange)
+	}
+	return v4Subnet, v6Subnet, nil
+}
+
+/*
+GCP hard-codes the amount of alias IPs that can be assigned to a NIC to 10 -
+independently of IP family, so we need to retrive the amount of alias IPs
+already in use by default and subtract from 10. See:
+https://cloud.google.com/vpc/docs/quota#per_instance .
+
+Note: there is also a global "alias IP per VPC quota", but OpenShift clusters on
+GCP seem to have that value defined to 15,000. So we can skip that.
```

Comment:
This may be because of the instance types or zones or whatever that we're using, which may not be the ones that everyone else is using?

(Also, `//` for comment)